### DISCUSSION TITLE
Google Chat bot integration

### DISCUSSION THREAD

**Discussion by anyulled** [jreleaser/jreleaser#227] [Ideas]
Hi! I've created a [fork](https://github.com/anyulled/jreleaser) to integrate Jreleaser with google chat to send messages as a bot via [incoming webhooks](https://developers.google.com/chat/how-tos/webhooks)

Would you be interested in adding this feature to JReleaser? What are the required steps to merge it into the repository?

**Comment by aalmiray** [comment]
This is fantastic @anyulled! I'd be great to have this feature added as an additional announcer.
A few things:

- do you agree to the terms of the Apache V2 license?
- can you certify all code (not copied from existing JRelease code) was written by you?
- I think the name of the announcer should be `googleChat` and not `google-chat`, to match the property name. 
- please adjust the authorship to your name, even if the code is copied from existing JReleaser code.
- please adjust all `@since` tags to match `0.5.0`.
- matching documentation entries would need to be added to https://github.com/jreleaser/jreleaser.github.io/ on the development branch. I can take care of that once the feature is merged if you don not want to deal with Antora and Asciidoc.

The first 2 questions are normally addressed by a Contributor License Agreement as well as the IP rights and copyright. I've yet to decide if JReleaser should/must have a CLA but at the very least the copyright notice must be amended as I don't have any intention to take away your rights (or from any other contributor) without a binding contract in place (such as CLA).

**Comment by anyulled** [comment]
Thanks for your reply.

- Yes, I agree to the terms of the Apache V2 license.
- Yes, I certify all code was written only by me.
- Thanks, I was thinking of changing it since I saw another plugin that uses the camel case.
- Sure, I'll check all the files that I modified.
- Yes, I'll update the Javadoc documentation
- I know a bit of Asciidoc but have no knowledge of Antora, I'll give it a try and ask you for support in case I get stuck.

**Reply by aalmiray** [reply]
Great! In the meantime I'll draft a CLA and work a way to get allow digital signatures to ease the process for contributors.

**Reply by aalmiray** [reply]
A CLA is now in place. I believe you'll be asked to digitally sign it when submitting a PR for the first time. Copyright notice has also been updated. I'd recommend rebasing from main before sending the PR. Thanks again!

**Reply by aalmiray** [reply]
Hi there, I'm thinking in posting the next release either this weekend or by June 30th at the latest. If the code for the PR is ready feel free to send it as it is; if the docs are missing no worries, I can take care of that 😄

**Reply by anyulled** [reply]
Hi there, right now I'm on holidays and I'll be back probably on Sunday. I'll give it another check then and send you the PR as soon as I can. Many thanks!

**Reply by aalmiray** [reply]
Sorry if I sounded pushy, no need to rush. Enjoy the time off!

**Comment by anyulled** [comment]
Not pushy at all, thanks for the information. I'll do my best to have it
merged before the next meetup with BarcelonaJug 😉

Anyul Rivas

**Reply by aalmiray** [reply]
Heads up, just added generic webhook support https://github.com/jreleaser/jreleaser/issues/237, I believe you'll have to rebase your work once again.

There's a small change in the `org.jreleaser.model.AbstractAnnouncer.asMap(Map<String, Object> props)` method, it now takes an second argument of type `boolean`.

Even though generic webhooks may cover Google ChatBot I'd still prefer explicit support for it, as existing announcers that rely on webhooks (Gitter, Slack, Discord, etc) will remain as well.

**Reply by anyulled** [reply]
Hi! understood, I'll do the rebase today and create the PR

**Reply by anyulled** [reply]
thanks for the update
